### PR TITLE
closes #2085: publish v2 quarkus-common jar to the maven repository

### LINKED ISSUE #2085: Publish `sgv2-quarkus-common` jar to Maven Central
While Stargate v1 publishes all artifacts as jars, even if many/most are not needed (Coordinators), Stargate v2 stops publishing of all APIs as jars: they are published as Docker images which makes more sense. But there is one component, `sgv2-quarkus-common` that is a library and will be needed by anyone implementing Stargate V2 APIs/Services.
It should be published as a jar, to Maven Central to be accessible by external developers.

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Publishes `svg2-quarkus-common` to the Maven repository:

* uses similar setup as the base pom
* however, with the newest plugin versions
* I had to do some gymnastics to enable only `svg2-quarkus-common` in the deploy profile

**Which issue(s) this PR fixes**:
Fixes #2085

### CODE REVIEW

**Comment by jeffreyscarpenter** [comment]
Just wondering if publishing the service Jars is such a bad thing, especially if we have to put in so much effort to avoid publishing them?

File: apis/pom.xml [hunk 1]
```diff
@@ -162,5 +174,111 @@
...
+              <serverId>ossrh</serverId>
+              <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+              <autoReleaseAfterClose>true</autoReleaseAfterClose>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
   </profiles>
+  <licenses>
```
Comment (tatu-at-datastax) [inline apis/pom.xml:266]: Maybe move "developers", "licenses" and "scm" up towards beginning of file.

**Review by tatu-at-datastax — APPROVED** [review]
LGTM

**Comment by tatu-at-datastax** [comment]
Looks like "Unit tests" fails with:

```
java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for io.stargate:sgv2-docsapi::jar:2.0.0-BETA-3-SNAPSHOTnull
```

so something missing/wrong in `pom.xml` or CI.

**Comment by ivansenic** [comment]
> Just wondering if publishing the service Jars is such a bad thing, especially if we have to put in so much effort to avoid publishing them?

Who would ever use this?

**Comment by ivansenic** [comment]
> Looks like "Unit tests" fails with:
> 
> ```
> java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for io.stargate:sgv2-docsapi::jar:2.0.0-BETA-3-SNAPSHOTnull
> ```
> 
> so something missing/wrong in `pom.xml` or CI.

Don't get this, I can not reproduce locally.. Doing exactly the same thing as in the CI.. I do assume it's something with the modules, or some caching issues maybe..

File: apis/pom.xml (hunk 1 above)
Comment (ivansenic) [reply apis/pom.xml:251]: where exactly would this fit?

**Comment by ivansenic** [comment]
@tatu-at-datastax seems that the modules in profiles was not working, so I reverted and added a `-pl sgv2-quarkus-common` to the github workflow.. Can you please re-check again?

File: apis/pom.xml (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/pom.xml:251]: Earlier after artifact and group ids, description; before dependencies and properties is typical.

But whatever, it's fine.

**Comment by tatu-at-datastax** [comment]
> @tatu-at-datastax seems that the modules in profiles was not working, so I reverted and added a `-pl sgv2-quarkus-common` to the github workflow.. Can you please re-check again?

Was just looking at CI, seems to pass now.

**Pull request merged**